Open Judge System/Data/OJS.Data.Models/Problem.cs

```diff
@@ -62,6 +62,8 @@ public class Problem : DeletableEntity, IOrderable
         [DefaultValue(false)]
         public bool ShowDetailedFeedback { get; set; }
 
+        public int DefaultSubmissionTypeId { get; set; }
```

Comment:
Why not nullable id? It is not required to have a default submission type. Also you can add navigational property for ease of use.

Reply:
Do we need navigational property for the defaultSubmissionType. We already have many to many relation and can extract it from there.

Reply:
The relation is already made, so the navigational property will be just for convenience and it's not bad to have it.

Reply:
Done